spec/models/observation_spec.rb

```diff
@@ -3572,151 +3667,193 @@ def stub_api_response_for_observation( o )
+    subject { observation.fields_addable_by? field_adder }
 
-    it "should be true for curators if curators preferred" do
-      c = make_curator
-      u = User.make!(:preferred_observation_fields_by => User::PREFERRED_OBSERVATION_FIELDS_BY_CURATORS)
-      o = Observation.make!(:user => u)
-      expect(o.fields_addable_by?(c)).to be true
-    end
+    context "for anyone else" do
+      it { is_expected.to be true }
 
-    it "should be true for curators by default" do
-      c = make_curator
-      u = User.make!
-      o = Observation.make!(:user => u)
-      expect(o.fields_addable_by?(c)).to be true
+      context "no editing preferred" do
+        let(:observer_preference) { User::PREFERRED_OBSERVATION_FIELDS_BY_OBSERVER }
+        let(:observer) { build_stubbed :user, preferred_observation_fields_by: observer_preference }
+
+        it { is_expected.to be false }
+      end
     end
 
-    it "should be false for curators if no editing preferred" do
-      c = make_curator
-      u = User.make!(:preferred_observation_fields_by => User::PREFERRED_OBSERVATION_FIELDS_BY_OBSERVER)
-      o = Observation.make!(:user => u)
-      expect(o.fields_addable_by?(c)).to be false
+    context "for nil user" do
+      let(:field_adder) { nil }
+
+      it { is_expected.to be false }
     end
 
-    it "should be false for everyone other than the observer if no editing preferred" do
-      other = User.make!
-      u = User.make!(:preferred_observation_fields_by => User::PREFERRED_OBSERVATION_FIELDS_BY_OBSERVER)
-      o = Observation.make!(:user => u)
-      expect(o.fields_addable_by?(other)).to be false
+    context "for curator" do
+      let(:field_adder) { build_stubbed :curator }
+
+      it { is_expected.to be true }
+
+      context "and curators preferred" do
+        let(:observer_preference) { User::PREFERRED_OBSERVATION_FIELDS_BY_CURATORS }
+        let(:observer) { build_stubbed :user, preferred_observation_fields_by: observer_preference }
+
+        it { is_expected.to be true }
+      end
+
+      context "and no editing preferred" do
+        let(:observer_preference) { User::PREFERRED_OBSERVATION_FIELDS_BY_OBSERVER }
+        let(:observer) { build_stubbed :user, preferred_observation_fields_by: observer_preference }
+
+        it { is_expected.to be false}
+      end
     end
 
-    it "should be true for the observer if no editing preferred" do
-      u = User.make!(:preferred_observation_fields_by => User::PREFERRED_OBSERVATION_FIELDS_BY_OBSERVER)
-      o = Observation.make!(:user => u)
-      expect(o.fields_addable_by?(u)).to be true
+    context "for observer" do
+      let(:field_adder) { observer }
+
+      context "and no editing preferred" do
+        let(:observer_preference) { User::PREFERRED_OBSERVATION_FIELDS_BY_OBSERVER }
+        let(:observer) { build_stubbed :user, preferred_observation_fields_by: observer_preference }
+
+        it { is_expected.to be true}
+      end
     end
   end
 
   describe "mappable" do
```

Comment:
This describe is a good example of what I've tried to break out. When testing the behavior of the method `calculate_mappable` we can use stubbed objects, however when introducing interactions such as "with a photo" or "should not be mappable if community disagrees with taxon", the specs continue to create fully persisted objects to maintain the integrative approach (if it's not already covered in request/controller specs).

Reply:
Seems prudent from a test performance perspective, and from the perspective of continuous integration. Definitely not how I think when I'm trying to do test-driven development, though, when I would never bother to test the framework, as you put it, but rather want to focus on testing my assumptions about the behavior of the code I'm about to write.